plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadataFactory.java

```diff
@@ -25,17 +26,21 @@
 {
     private final JdbcClient jdbcClient;
     private final boolean allowDropTable;
+    private final Set<SessionPropertiesProvider> sessionPropertiesProviders;
 
     @Inject
-    public JdbcMetadataFactory(JdbcClient jdbcClient, JdbcMetadataConfig config)
+    public JdbcMetadataFactory(JdbcClient jdbcClient, JdbcMetadataConfig config, Set<SessionPropertiesProvider> sessionPropertiesProviders)
     {
         this.jdbcClient = requireNonNull(jdbcClient, "jdbcClient is null");
         requireNonNull(config, "config is null");
         this.allowDropTable = config.isAllowDropTable();
+        this.sessionPropertiesProviders = requireNonNull(sessionPropertiesProviders, "sessionPropertiesProviders is null");
     }
 
     public JdbcMetadata create()
     {
-        return new JdbcMetadata(new TransactionCachingJdbcClient(jdbcClient, new Duration(1, TimeUnit.DAYS)), allowDropTable);
+        return new JdbcMetadata(
+                new CachingJdbcClient(jdbcClient, sessionPropertiesProviders, new Duration(1, TimeUnit.DAYS), true),
```

Comment:
`TransactionCachingJdbcClient` (that used to be used here), did not provide session property providers:

```
  // session stays the same per transaction, therefore session properties don't need to	
        // be a part of cache keys in CachingJdbcClient
```

did anything change in this regard here?